components/prize_pool/commons/prize_pool_placement.lua

```diff
@@ -271,6 +269,25 @@ function Placement:_parseOpponents(args)
 	end)
 end
 
+function Placement:_shouldAddTbdOpponent(opponentIndex, place)
+	local slotSize
+	if place then
+		slotSize = self.placeEnd - self.placeStart + 1
+	end
+
+	if
+		-- we want at least 1 opponent present for any placement
+		opponentIndex == 1
+		-- also fill up if the fillPlaceRange option is enabled and we have an entered placeRange
+		-- and we do not yet have reached the number of Opponents for that range
+		or (self.parent.options.fillPlaceRange and place and opponentIndex <= slotSize)
+	then
+		return true
+	end
+
+	return false
+end
```

Comment:
```suggestion
function Placement:_shouldAddTbdOpponent(opponentIndex, place)
	-- We want at least 1 opponent present for all placement
	if opponentIndex == 1 then
		return true
	end

	-- If the fillPlaceRange option is enabled and add TBD's for the entire slot
	local slotSize = self.placeEnd - self.placeStart + 1
	if self.parent.options.fillPlaceRange and opponentIndex <= slotSize) then
		return true
	end

	return false
end
```

Reply:
would error

self.placeEnd and self.placeStart are only present at this point if place is set

Reply:
the following could work (not yet tested)
```suggestion
function Placement:_shouldAddTbdOpponent(opponentIndex, place)
	-- We want at least 1 opponent present for all placements
	if opponentIndex == 1 then
		return true
	end
	-- If the fillPlaceRange option is enabled and we have a placeRange is given add TBD's for the entire slot
	local slotSize = place and (self.placeEnd - self.placeStart + 1) or 0
	if self.parent.options.fillPlaceRange and opponentIndex <= slotSize then
		return true
	end
	return false
end
```

Reply:
an alternative:
```suggestion
function Placement:_shouldAddTbdOpponent(opponentIndex, place)
	-- We want at least 1 opponent present for all placements
	if opponentIndex == 1 then
		return true
	end
	-- If the fillPlaceRange option is disabled or we do not have a give placeRange do not fill up further
	if not self.parent.options.fillPlaceRange or not place then
		return false
	end
	-- Only fill up further with TBD's if there is free space in the placeRange/slot
	local slotSize = place and (self.placeEnd - self.placeStart + 1) or 1
	if opponentIndex <= slotSize then
		return true
	end
	return false
end
```

Reply:
282 `place` shouldn't be needed as that's checked on 278

Reply:
> 282 `place` shouldn't be needed as that's checked on 278

adjusted (bad copy paste)

Reply:
I like the alternative one due to the clarity of it